### DISCUSSION TITLE
Only retrieving listed package versions

### DISCUSSION THREAD

**Discussion by LPeter1997** [NuGet/Home#11152] [Q&A] [answered]
I'm using the NuGet Client SDK to unlist my previous nightly builds before uploading the new one. Tonight I've hit a quota limit, so my CI and the nightly publish failed. I'm managing around 20 packages, which means around 20 new nightly versions each night.

I could not find any way to only request for the listed versions using `GetAllVersionsAsync`.

How could I solve this? Ideally, I could just specify a filter for getting the package versions, as I really only care about ones that are listed and are pre-release (or end with my given suffix).

**Comment by nkolev92** [comment, marked as answer]
Please refer to https://docs.microsoft.com/en-us/nuget/reference/nuget-client-sdk#get-package-metadata
